xapian-glib/xapian-database.cc

```diff
@@ -700,9 +685,26 @@ xapian_database_compact_to_fd (XapianDatabase             *self,
 #endif
 }
 
-XapianDatabaseFlags
+int
 xapian_database_get_flags (XapianDatabase *self)
 {
   XapianDatabasePrivate *priv = XAPIAN_DATABASE_GET_PRIVATE (self);
+  int db_flags = 0;
+
+  if (priv->flags & XAPIAN_DB_NO_SYNC)
+    db_flags |= Xapian::DB_NO_SYNC;
+  if (priv->flags & XAPIAN_DB_FULL_SYNC)
+    db_flags |= Xapian::DB_FULL_SYNC;
+  if (priv->flags & XAPIAN_DB_DANGEROUS)
+    db_flags |= Xapian::DB_DANGEROUS;
+  if (priv->flags & XAPIAN_DB_RETRY_LOCK)
+    db_flags |= Xapian::DB_RETRY_LOCK;
+  if (priv->flags & XAPIAN_DB_BACKEND_GLASS)
+    db_flags |= Xapian::DB_BACKEND_GLASS;
+  if (priv->flags & XAPIAN_DB_BACKEND_CHERT)
+    db_flags |= Xapian::DB_BACKEND_CHERT;
+  if (priv->flags & XAPIAN_DB_BACKEND_STUB)
+    db_flags |= Xapian::DB_BACKEND_STUB;
+
   return priv->flags;
```

Comment:
Yes, it should; clearly the test suite is not good enough, because it didn't catch it.